### PR TITLE
Fxd point revisions 

### PR DESCRIPTION
**Pull request by yoonachang**
fixed the std.lib, added data for the test, added python code to convert decimal numbers to use fixedpoints

### CODE REVIEW

File: fixedpoint/fixedpoint.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import numpy as np
+
+def decimal_to_fixed_p (num, width, int_bit, frac_bit):
+    '''
```
Comment (rachitnigam) [inline fixedpoint/fixedpoint.py:4]: Throw an error if the number is not representable using the number of fractionals bits you have. For example 10.8 cannot be represented using 2 fractional bits,

File: primitives/std.lib [hunk 2]
```diff
@@ -820,26 +776,12 @@ primitive fixed_p_std_mult[p_width, p_int_width, p_fract_width](left: p_width, r
...
         output logic [width-1:0] out);
 
-        logic [width-1:width-int_width] l_int, r_int, out_int, scaled_int; 
-        logic [width-int_width-1:0] l_scale, r_scale, new_scale;
+        logic [2*width-2:0] result;
 
-        assign {l_int, l_scale} = left;
-        assign {r_int, r_scale} = right;
+        assign result = left * right;
+        assign out = result[width+fract_width-1:fract_width];
```
Comment (rachitnigam) [inline primitives/std.lib:782]: Add a comment to explain this slicing.

File: fixedpoint/fixedpoint.py (hunk 1 above)
Comment (yoonachang) [reply fixedpoint/fixedpoint.py:4]: Would raising exceptions also work?

Comment (sgpthomas) [reply fixedpoint/fixedpoint.py:4]: yeah that'd work

Comment (yoonachang) [reply fixedpoint/fixedpoint.py:4]: Do I need additional checks to be able to merge?

Comment (rachitnigam) [reply fixedpoint/fixedpoint.py:4]: Don't know why they are not triggering. I'll merge.

**Pull request merged**